src/libopensc/apdu.c

```diff
@@ -559,6 +559,14 @@ int sc_transmit_apdu(sc_card_t *card, sc_apdu_t *apdu)
 		return r;
 	}
 
+#if ENABLE_SM
+	if (card->sm_ctx.sm_mode == SM_MODE_TRANSMIT
+			&& (apdu->flags & SC_APDU_FLAGS_CHAINING) != 0
+			&& (apdu->flags & SC_APDU_FLAGS_SM_CHAINING) != 0) {
+		sc_log(card->ctx,"Let SM do the chaining");
+		r = sc_transmit(card, apdu);
+	} else
+#endif
```

Comment:
See 7cf675b83196c9fdac595f1078b2d8691d65a55c This was one of the first issues I ran into with NIST SM.

    Allow driver to tell apdu.c to pass to sm_get_apdus to have it do
    the chaining. Without this change, apdu.c will do the command chaining
    and pass each chained apdu to SM. This may be OK for some,
    but not PIV which uses short APDUs and encryption and MAC for data
    before doing the chaining.

    The driver sets SC_APDU_FLAGS_SM_CHAINING flag and SC_APDU_FLAGS_CHAINING.
    which causes apdu.c to pass the plain adpu to card->sm_ctx.ops.get_sm_apdu.

    Without the SC_APDU_FLAGS_SM_CHAINING the flow of SM is not changed.